bip-musig2/gen_vectors_helper.py

```diff
@@ -17,6 +17,34 @@ def gen_key_agg_vectors():
     print("  pubkey:", pk.hex().upper())
     print("  tweak: ", tweak.hex().upper())
 
+def check_sign_verify_vectors():
+    with open(os.path.join(sys.path[0], 'sign_verify_vectors.json')) as f:
+        test_data = json.load(f)
+    X = fromhex_all(test_data["pubkeys"])
+    pnonce = fromhex_all(test_data["pnonces"])
+    aggnonces = fromhex_all(test_data["aggnonces"])
+    msgs = fromhex_all(test_data["msgs"])
+
+    valid_test_cases = test_data["valid_test_cases"]
+    for (i, test_case) in enumerate(valid_test_cases):
+        pubkeys = [X[i] for i in test_case["key_indices"]]
+        pubnonces = [pnonce[i] for i in test_case["nonce_indices"]]
+        aggnonce = aggnonces[test_case["aggnonce_index"]]
+        assert nonce_agg(pubnonces) == aggnonce
+        msg = msgs[test_case["msg_index"]]
+        signer_index = test_case["signer_index"]
+        expected = bytes.fromhex(test_case["expected"])
+
+        session_ctx = SessionContext(aggnonce, pubkeys, [], [], msg)
+        (Q, _, _, _, R, _) = get_session_values(session_ctx)
+        # Make sure the vectors include tests for both variants of Q and R
+        if i == 0:
+           assert has_even_y(Q) and not has_even_y(R)
+        if i == 1:
+           assert not has_even_y(Q) and has_even_y(R)
+        if i == 2:
+           assert has_even_y(Q) and has_even_y(R)
```

Comment:
Nit: Is it worth making this check less sensitive by ignoring the order of tests? For example something like this:

```python
qr_parity = set()
...
    qr_parity.add(2 * has_even_y(Q) + has_even_y(R))
...
assert len(qr_parity) == 3
```

Reply:
I think sensitivity is an advantage here. That way we know exactly which behavior is supposed to be exhibited by each test.